static/js/buttons/instances/CreateSnapshotBtn.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FC, useState } from "react";
+import { Button } from "@canonical/react-components";
+import CreateSnapshotForm from "../../modals/CreateSnapshotForm";
+import { LxdInstance } from "../../types/instance";
+import { NotificationHelper } from "../../types/notification";
+
+interface Props {
+  instance?: LxdInstance;
+  notify: NotificationHelper;
+}
+
+const CreateSnapshotBtn: FC<Props> = ({ instance, notify }) => {
+  const [isModal, setModal] = useState(false);
+
+  return (
+    <>
+      {instance && isModal && (
```

Comment:
I think checking that `instance` is not `undefined` should not be done here, as we shouldn't even have this rendered if `instance` is not defined. We also do the same in other places I think, check in the parent and then pass a clean, "defined" prop to the children.